dockerfiles/feature-support/hsm-patches/0001-Introduce-generic-keypair-generation-interface-and-e.patch

```diff
@@ -417,36 +443,31 @@ index ec7f279..57cba43 100644
 +	curve_obj = OBJ_nid2obj(curve_nid);
 +	if (!curve_obj)
 +		return -1;
-+	ecdsa_params_len = i2d_ASN1_OBJECT(curve_obj, NULL);
-+	ecdsa_params = (unsigned char *)OPENSSL_malloc(ecdsa_params_len);
-+	if (!ecdsa_params)
++	ec_params_len = i2d_ASN1_OBJECT(curve_obj, NULL);
++	ec_params = (unsigned char *)OPENSSL_malloc(ec_params_len);
++	if (!ec_params)
 +		return -1;
-+	tmp = ecdsa_params;
++	tmp = ec_params;
 +	i2d_ASN1_OBJECT(curve_obj, &tmp);
```

Comment:
readability: Why do we pass tmp here and not ec_params? I just looked up what I believe to be the reason for this in OpenSSLs documentation. tmp is a bad variable name if the purpose isn't obvious and it deserves a comment here to explain whats going on with strange OpenSSL API.

return value check?

Reply:
Documented and return value checked